driver/src/util.rs

```diff
@@ -38,18 +38,31 @@ pub fn hash_consistency_check(
     Ok(())
 }
 
-pub fn can_process<C>(
+pub enum ProcessingState {
+    TooEarly,
+    AcceptsBids,
+    AcceptsSolution,
+}
+
+pub fn batch_processing_state<C>(
     slot: U256,
     contract: &C,
-    creation_block: &Fn(U256) -> Result<U256, DriverError>,
-) -> Result<bool, DriverError>
+    creation_block_time: &Fn(U256) -> Result<U256, DriverError>,
+) -> Result<ProcessingState, DriverError>
 where
     C: SnappContract,
 {
-    let slot_creation_block = creation_block(slot)?;
-    if slot_creation_block == U256::zero() {
-        return Ok(false);
+    let slot_creation_block_time = creation_block_time(slot)?;
+    if slot_creation_block_time == U256::zero() {
+        return Ok(ProcessingState::TooEarly);
+    }
+
+    let current_block_time = contract.get_current_block_timestamp()?;
+    if slot_creation_block_time + 360 < current_block_time {
```

Comment:
I would consider using a centralized set of constants for this (avoid magic numbers)